#163446097 Add route to edit party name

Simplemart17 opened the pull request:
#### What does this PR do?
it adds route to edit the name of a specific political party
it add unit-tests for the API endpoints

#### Description of Task to be completed?
* Have the following api endpoints working
```POST /api/v1/party/```
```GET /api/v1/party```
```GET/api/v1/party/:id```
```PATCH/api/v1/party/:id/name```

#### How should this be manually tested?
*Clone this repository and run ```npm install``` followed by ```npm start``` to start the server*
* Use Postman to test the endpoints above and use the following header
*key*: Content-type
*value*: Application/json

#### What are the relevant pivotal tracker stories?
[#163446097](https://www.pivotaltracker.com/story/show/163446097)

Simplemart17 committed:
feat(api): add party patch route

add endpoint route to edit the name of the party

[Finishes #163446097]

Simplemart17 committed:
feat(api): add unit test for patch route

add unit test for the edit party name route

[Finishes #163446097]

andela-gike commented on app/controllers/partyController.js:
```diff
@@ -40,6 +41,43 @@ class PartyController {
+    partydb.splice(itemIndex, 1, data);
+
+    return res.status(201).json({
+      message: 'Party name was successfully edited!',
```
Confirm the status code for updating, and your success message is specific to only the name, would some only be able to update a name and nothing else. So the party hq address cannot change ever.

Simplemart17 replied:
I have updated the status code and return specific status message for the edited party.
However, for updating other information about the party, following the provision of the rubric, only the name was specified to be patched and nothing else.
Thank you.

andela-gike replied:
ok

Simplemart17 committed:
feat(api): modify the patch party endpoint

change the status code for the endpoint
modify the status message to specific to the name of the edited party

[Finishes #163446097]
